### PR TITLE
Cleanup Sonarcloud issues

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

**Comment by FFroehlich** [comment]
Should fix most of the bugs/issues that sonarcloud marks as "blocker", bit disappointed that sonarcloud doesn't track this in any meaningfull way.

**Comment by dweindl** [comment]
> bit disappointed that sonarcloud doesn't track this in any meaningfull way.

Same here. Should show stats for removed code smells. Feature request? :)

File: include/amici/defines.h [hunk 1]
```diff
@@ -18,37 +18,37 @@ constexpr double pi = 3.14159265358979323846;
...
+
+constexpr int AMICI_PREEQUILIBRATE=           -1;
 
 #ifndef booleantype
 #define booleantype int
 #endif
 
 #ifndef FALSE
-#define FALSE 0
+constexpr int FALSE=0;
```
Comment (dweindl) [inline include/amici/defines.h:47]: With the replacements below, is this still being used anywhere?

Comment (FFroehlich) [reply include/amici/defines.h:47]: unfortunately yes, the solver has to use it in a couple of places.

Comment (FFroehlich) [reply include/amici/defines.h]: done, we are gonna wait for CI for a while anyways :D

**Pull request merged**